x/xio/types.go

```diff
@@ -63,20 +70,20 @@ type ReaderSliceOfSlicesIterator interface {
 	// Next moves to the next item
 	Next() bool
 
-	// CurrentLen returns the current slice of readers
-	CurrentLen() int
+	// Current returns the current length, start, and end time
+	Current() (length int, start, end time.Time)
```

Comment:
hm did a quick search of all the places that use this method in the code ([sourcegraph](https://sourcegraph.com/github.com/m3db/m3db@312f5cca55ac3a64c32febd3a4798fa12d7e538d/-/blob/x/xio/types.go#L74:2$references) is dope af), the new version is only used in tests. i.e. We could retain `CurrentLen()` (like the original) and it'd be fine.

Reply:
The tool doesn't seem to show this usage for some reason
https://github.com/m3db/m3db/pull/532/files/312f5cca55ac3a64c32febd3a4798fa12d7e538d#diff-d28c20c8fa3884aa01db69b76c5d300aR59